README.md

```diff
@@ -1,53 +1,76 @@
-2. Run `docker-compose up`. This will start Redis server on docker
+```sh
+python3 -m venv climateconnect_env
+cd climateconnect_env
+source bin/activate
+```
+
+Then clone the GitHub repository
+
+```sh
+git clone https://github.com/climateconnect/climateconnect
+```
 
 ### Backend
 
-Once you clone the repository start backend server.
+After you've cloned the repository, we can set up the local Redis server and backend.
+
+Note: we use Python 3, so for all instructions we insume `python` means `python3`.
+
+#### First Time Setup
+1.  Go to backend directory: `cd backend`
+1.  Run `pip install -r requirements.txt` to install all backend libararies.
+1.  Create `.backend_env` to set environment variables.
+    - You can find up to date sample env variables in [`backend/local-env-setup.md`](https://github.com/climateconnect/climateconnect/blob/master/backend/local-env-setup.md).
+    - For the Django `SECRET_KEY`, run `openssl rand -base64 32` to create a 32 char random secret.
+    - Note that you'll also need a [Django `SECRET_KEY`](https://docs.djangoproject.com/en/3.1/ref/settings/#std:setting-SECRET_KEY).
+1.  Run `python manage.py migrate` to run Django migrations.
+    - Note: This command is used for when you first start, or whenever you are adding or updating database models.
+
+
+#### Continual Development
+
+1.  Run `docker-compose up`. This will start a Redis server on Docker.
+1.  Run `python manage.py migrate` to run Django migrations whenever you are adding or updating database model.s
+1. Run server using `python manage.py runserver`.
+
+
+#### Testing
+If you want to run the test suite, use:
+
+```sh
+python manage.py test
+```
 
-1.  Go to backend directory `cd backend`
-2.  Run `pip install -r requirements.txt` to install all backend libararies.
-3.  Create `.backend_env` to include backend environment variables. You can find upto date sample env variables in [`backend/local-env-setup.md`](https://github.com/climateconnect/climateconnect/blob/master/backend/local-env-setup.md) file.
-4.  Run `python manage.py migrate` to run django migrations. Note: This command is used for
-    when you first start or whenever you are adding or updating database models.
-5.  Run server using `python manage.py runserver`.
-6.  If you want to run tests suite use this command: `python manage.py test`. If you want to run a
-    specific test file or test class run this command:
-    `python manage.py test <file_path> or <file_path + class_name>`
+Run a specific test file or test class:
+```sh
+python manage.py test <file_path> or <file_path + class_name>
+```
 
 ### Frontend
 
 1. `cd frontend`
 2. Run `yarn install` to download all npm packages
-3. Add `.env` file for frontend environment variables. You can find which env variables you have to set in `/frontend/next.config.js/`
+3. Add a `.env` file for frontend environment variables. You can find which env variables you have to set in `/frontend/next.config.js/`
```

Comment:
Maybe we could add a link to the file, as we did above in the backend section. The link would be https://github.com/climateconnect/climateconnect/blob/master/frontend/next.config.js